src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2358,6 +2365,35 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         return _core->TaskbarProgress();
     }
 
+    void TermControl::BellLightOn()
+    {
+        if (!_bellLightTimer)
+        {
+            // Start the timer, when the timer ticks we switch off the light
+            DispatcherTimer invertTimer;
+            invertTimer.Interval(std::chrono::milliseconds(TerminalWarningBellInterval));
+            invertTimer.Tick({ get_weak(), &TermControl::_BellLightOff });
+            invertTimer.Start();
+            _bellLightTimer.emplace(std::move(invertTimer));

```

Comment:
BTW If you want to you can replace the entire `DispatchTimer` logic with a potentially simpler coroutine solution... Something like this (untested):

```cpp
winrt::fire_and_forget TermControl::BellLightOn()
{
    auto weakSelf = get_weak();
    
    VisualBellLight::SetIsTarget(RootGrid(), true);
    BellLight().CompositionLight().StartAnimation(L"Intensity", _bellLightAnimation);
    
    winrt::apartment_context ui_thread;
    co_await winrt::resume_after(TerminalWarningBellInterval);
    co_await ui_thread;
    
    if (auto self{ weakThis.get() })
    {
        VisualBellLight::SetIsTarget(self.RootGrid(), false);
    }
}
```

Reply:
I'd like to keep it this way for now so we have consistency with `cursorTimer` and `blinkTimer`!